tensorflow/python/ops/array_ops.py

```diff
@@ -1817,8 +1817,8 @@ def _DepthToSpaceShape(op):
       [input_shape[0], height, width, new_depth])]
 
 
-def one_hot(indices, depth, on_value=1, off_value=0,
-            axis=None, dtype=dtypes.float32, name=None):
+def one_hot(indices, depth, on_value=1.0, off_value=0.0,
```

Comment:
set default on_value and off_value = None; that makes the checks below much easier.